src/app/public/modules/repeater/repeater-item.component.ts

```diff
@@ -189,6 +239,72 @@ export class SkyRepeaterItemComponent implements OnDestroy, OnInit {
     this.inlineFormClose.emit(inlineFormCloseArgs);
   }
 
+  public moveToTop(event: Event): void {
+    event.stopPropagation();
+    this.adapterService.moveItemUp(this.elementRef, true);
+    (<HTMLElement> event.target).focus();
+  }
+
+  public handleKeyboardEvent(event: KeyboardEvent): void {
+    let key = event.key.toLowerCase();
+    if (key === ' ' || key === 'enter') {
+      this.keyboardDraggingEnabled = !this.keyboardDraggingEnabled;
+      this.reorderSteps = 0;
+
+      if (this.keyboardDraggingEnabled) {
+        this.reorderState = this.reorderStateDescription;
+      } else {
+        this.reorderState = this.reorderFinishText + ' ' + (this.reorderCurrentIndex + 1)  + ' ' + this.reorderInstructions;
+      }
+
+      event.stopPropagation();
+    } else if (key === 'escape') {
+      this.keyboardDraggingEnabled = false;
+
+      if (this.reorderSteps < 0) {
+        this.adapterService.moveItemDown(this.elementRef, Math.abs(this.reorderSteps));
+      } else if (this.reorderSteps > 0) {
+        this.adapterService.moveItemUp(this.elementRef, false, this.reorderSteps);
+      }
+
+      this.reorderButtonLabel = this.reorderCancelText + ' ' + this.reorderInstructions;
+
+      (<HTMLElement> event.target).focus();
+
+      event.stopPropagation();
+    } else if (this.keyboardDraggingEnabled && key.startsWith('arrow')) {
+      let direction = event.key.toLowerCase().replace('arrow', '');
+      if (direction === 'up') {
+        this.reorderCurrentIndex = this.adapterService.moveItemUp(this.elementRef);
+        this.reorderSteps--;
+        this.grabHandle.nativeElement.focus();
+        this.keyboardDraggingEnabled = true;
+        this.reorderButtonLabel = this.reorderMovedText + ' ' + (this.reorderCurrentIndex + 1);
+      } else if (direction === 'down') {
+        this.reorderCurrentIndex = this.adapterService.moveItemDown(this.elementRef);
+        this.reorderSteps++;
+        this.grabHandle.nativeElement.focus();
+        this.keyboardDraggingEnabled = true;
+        this.reorderButtonLabel = this.reorderMovedText + ' ' + (this.reorderCurrentIndex + 1);
+      }
+
+      event.stopPropagation();
```

Comment:
Could we move `event.stopPropagation()` from all three `if` blocks to just one at the end?

Reply:
Yup. Done